tests/Http/CurlRequestHandlerTest.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace YotiTest\Http;
+
+use Yoti\YotiClient;
+use Yoti\Http\CurlRequestHandler;
+use Yoti\Http\RequestSigner;
+use YotiTest\TestCase;
+use Yoti\Http\Payload;
+use Yoti\Entity\Country;
+use Yoti\Entity\AmlAddress;
+use Yoti\Entity\AmlProfile;
+
+class CurlRequestHandlerTest extends TestCase
+{
+    public $pem;
+    /**
+     * @var CurlRequestHandler
+     */
+    public $requestHandler;
+    /**
+     * @var Payload
+     */
+    public $payload;
+
+    public $requestSigner;
+
+    public $postMethod = CurlRequestHandler::METHOD_POST;
+
+    public function setup()
+    {
+        $endpoint = '/aml-check';
+        $amlAddress = new AmlAddress(new Country('GBR'));
+        $amlProfile = new AmlProfile('Edward Richard George', 'Heath', $amlAddress);
+        $this->payload = new Payload($amlProfile->getData());
+        $this->postMethod = CurlRequestHandler::METHOD_POST;
+
+        $this->pem = file_get_contents(PEM_FILE);
+
+        $this->requestSigner = new RequestSigner(
+            $this->requestHandler,
+            $endpoint,
+            'POST'
+        );
+
+        $this->requestHandler = new CurlRequestHandler(
+            YotiClient::DEFAULT_CONNECT_API,
+            $this->pem,
+            SDK_ID,
+            'PHP'
+        );
+    }
+
+    public function testGetPem()
+    {
+        $this->assertEquals($this->pem, $this->requestHandler->getPem());
+    }
+
+    public function testGetSdkId()
+    {
+        $this->assertEquals(SDK_ID, $this->requestHandler->getSdkId());
+    }
+
+    public function testInvalidHttpMethod()
+    {
+        $this->expectException('\Yoti\Exception\RequestException');
+        $this->requestHandler->sendRequest(
+            '/aml-check',
+            'InvalidHttpMethod',
+            $this->payload
+        );
+    }
+
+    public function testSendRequestExists()
+    {
+        $this->assertTrue(method_exists($this->requestHandler, 'sendRequest'));
+    }
+
+    public function testGenerateRequestHeaders()
+    {
+        $signedData = RequestSigner::signRequest(
+            $this->requestHandler,
+            '/aml-check',
+            'POST',
+            $this->payload
+        );
+        $signedMessage = $signedData[RequestSigner::SIGNED_MESSAGE_KEY];
+        $requestHeaders = $this->invokeMethod(
+            $this->requestHandler,
+            'generateRequestHeaders',
```

Comment:
If I understand well, here you are using reflection to test a private method.
I wonder if you could have tested `sendRequest` instead and mocked curl.
I've seen here how they mock it (wrap it into a class) https://stackoverflow.com/questions/7911535/how-to-unit-test-curl-call-in-php 
If you mock curl, then you can verify that the request contains the headers that you expect
when the wrapper of `curl_exec` is called.

Reply:
Ok will try that.